// ng-block-ui/bundles/umd/lib/components/block-ui-content/block-ui-content.component.js
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
(function (factory) {
    if (typeof module === "object" && typeof module.exports === "object") {
        var v = factory(require, exports);
        if (v !== undefined) module.exports = v;
    }
    else if (typeof define === "function" && define.amd) {
        define(["require", "exports", "@angular/core", "rxjs/add/operator/map", "../../services/block-ui.service", "../../constants/block-ui-actions.constant", "../../constants/block-ui-default-name.constant", "./block-ui-content.component.style", "./block-ui-content.component.template"], factory);
    }
})(function (require, exports) {
    "use strict";
    Object.defineProperty(exports, "__esModule", { value: true });
    var core_1 = require("@angular/core");
    require("rxjs/add/operator/map");
    var block_ui_service_1 = require("../../services/block-ui.service");
    var block_ui_actions_constant_1 = require("../../constants/block-ui-actions.constant");
    var block_ui_default_name_constant_1 = require("../../constants/block-ui-default-name.constant");
    var block_ui_content_component_style_1 = require("./block-ui-content.component.style");
    var block_ui_content_component_template_1 = require("./block-ui-content.component.template");
    var BlockUIContentComponent = (function () {
        function BlockUIContentComponent(blockUI) {
            this.blockUI = blockUI;
            this.name = block_ui_default_name_constant_1.BlockUIDefaultName;
            this.active = false;
        }
        BlockUIContentComponent.prototype.ngOnInit = function () {
            this.blockUISubscription = this.subscribeToBlockUI(this.blockUI.observe());
        };
        BlockUIContentComponent.prototype.subscribeToBlockUI = function (blockUI$) {
            var _this = this;
            return blockUI$
                .map(function (event) { return _this.onDispatchedEvent(event); })
                .subscribe();
        };
        BlockUIContentComponent.prototype.onDispatchedEvent = function (event) {
            switch (event.action) {
                case (block_ui_actions_constant_1.BlockUIActions.START):
                    this.onStart(event);
                    break;
                case (block_ui_actions_constant_1.BlockUIActions.STOP):
                case (block_ui_actions_constant_1.BlockUIActions.RESET):
                    this.onStop(event);
                    break;
            }
        };
        BlockUIContentComponent.prototype.onStart = function (event) {
            if (event.name === this.name) {
                this.active = true;
                this.message = event.message;
            }
        };
        BlockUIContentComponent.prototype.onStop = function (event) {
            if (event.name === this.name || event.action === block_ui_actions_constant_1.BlockUIActions.RESET) {
                this.active = false;
            }
        };
        BlockUIContentComponent.prototype.ngOnDestroy = function () {
            this.blockUISubscription.unsubscribe();
        };
        return BlockUIContentComponent;
    }());
    __decorate([
        core_1.Input(),
        __metadata("design:type", String)
    ], BlockUIContentComponent.prototype, "name", void 0);
    __decorate([
        core_1.Input('message'),
        __metadata("design:type", String)
    ], BlockUIContentComponent.prototype, "defaultMessage", void 0);
    BlockUIContentComponent = __decorate([
        core_1.Component({
            selector: 'block-ui-content',
            template: block_ui_content_component_template_1.template,
            styles: [block_ui_content_component_style_1.styles],
            encapsulation: core_1.ViewEncapsulation.None
        }),
        __metadata("design:paramtypes", [block_ui_service_1.BlockUIService])
    ], BlockUIContentComponent);
    exports.BlockUIContentComponent = BlockUIContentComponent;
});
